File: CS/WindowsFormsApplication28/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraVerticalGrid.Events;

namespace WindowsFormsApplication28
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			this.carsTableAdapter.Fill(this.carsDBDataSet.Cars);
		}

		private void vGridControl1_CustomUnboundData(object sender, CustomDataEventArgs e)
		{
			if ( e.Row != editorRow1 )
				return;

			if ( e.IsGetData )
				e.Value = "Caption " + e.ListSourceRowIndex;
		}
	}
}

File: VB/WindowsFormsApplication28/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraVerticalGrid.Events

Namespace WindowsFormsApplication28
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Me.carsTableAdapter.Fill(Me.carsDBDataSet.Cars)
		End Sub

		Private Sub vGridControl1_CustomUnboundData(ByVal sender As Object, ByVal e As CustomDataEventArgs) Handles vGridControl1.CustomUnboundData
			If e.Row IsNot editorRow1 Then
				Return
			End If

			If e.IsGetData Then
				e.Value = "Caption " & e.ListSourceRowIndex
			End If
		End Sub
	End Class
End Namespace
